File: examples/Python.ExampleApplication/Hooks/pulled/data_source_connection_pulled.py
from typing import Optional

from tableau_migration import (
    ContentItemPulledContext,
    ContentItemPulledHookBase,
    FilterStatus,
    IPublishableDataSource)

class DataSourceConnectionPulled(ContentItemPulledHookBase[IPublishableDataSource]):

    def execute(self, ctx: ContentItemPulledContext[IPublishableDataSource]) -> Optional[ContentItemPulledContext[IPublishableDataSource]]:
        if any(c.type.casefold() == "postgres".casefold() for c in ctx.pulled_item.connections):
            ctx.status = FilterStatus.CASCADE_SKIP

        return ctx

File: examples/Csharp.ExampleApplication/Hooks/Pulled/DataSourceConnectionPulled.cs
using System;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Tableau.Migration.Content;
using Tableau.Migration.Engine.Hooks.Filters;
using Tableau.Migration.Engine.Hooks.Pulled;

namespace Csharp.ExampleApplication.Hooks.Pulled
{
    #region class
    public class DataSourceConnectionPulled : ContentItemPulledHookBase<IPublishableDataSource>
    {
        public override Task<ContentItemPulledContext<IPublishableDataSource>?> ExecuteAsync(ContentItemPulledContext<IPublishableDataSource> ctx, CancellationToken cancel)
        {
            if (ctx.PulledItem.Connections.Any(c => string.Equals(c.Type, "postgres", StringComparison.OrdinalIgnoreCase)))
            {
                ctx.Status = FilterStatus.CascadeSkip;
            }

            return ctx.ToTask();
        }
    }
    #endregion
}
